flang/runtime/sum.cpp

```diff
@@ -188,7 +188,8 @@ void RTDEF(CppSumComplex16)(CppTypeFor<TypeCategory::Complex, 16> &result,
 void RTDEF(SumDim)(Descriptor &result, const Descriptor &x, int dim,
     const char *source, int line, const Descriptor *mask) {
   TypedPartialNumericReduction<IntegerSumAccumulator, RealSumAccumulator,
-      ComplexSumAccumulator>(result, x, dim, source, line, mask, "SUM");
+      ComplexSumAccumulator, /*MIN_REAL_KIND=*/4>(
```

Comment:
Does it mean that `SumReal4` accumulator and the accumulator for `SumDim` of `REAL(4)` will use different intermediate result data type?

Reply:
They should both use 32-bit floats with Kahan summation.  I'll double-check them and confirm.

Reply:
I need to also change the types in real/complex * 4 total summation; will do.